enclave-runtime/src/lib.rs

```diff
@@ -759,32 +768,29 @@ where
 		)));
 	}
 
-	prepare_and_send_xts_and_block::<_, SignedSidechainBlock, _, _>(
-		validator,
-		on_chain_ocall_api,
-		calls,
-		Default::default(),
-		nonce,
-	)
+	prepare_and_send_xts::<_, _, _, Signer>(validator, signer, on_chain_ocall_api, calls, nonce)
 }
 
-fn prepare_and_send_xts_and_block<Block, SB, V, OCallApi>(
+fn prepare_and_send_xts<Block, V, OCallApi, Signer>(
 	validator: &mut V,
+	signer: Signer,
 	ocall_api: &OCallApi,
 	calls: Vec<OpaqueCall>,
-	blocks: Vec<SB>,
-	nonce: &mut u32,
-) -> Result<()>
+	nonce: u32,
+) -> Result<u32>
```

Comment:
This function now only prepares and sends extrinsics. Does not do anything sidechain blocks related anymore.

In the next refactoring step I'll extract this function in to a crate. Here I'd like to have you opinions: Is there an existing crate where you think this function would fit in nicely? Or do I create a new crate for it?

Reply:
We have a Teerex-storage: https://github.com/integritee-network/worker/tree/master/core-primitives/teerex-storage

With the current naming, it does not fit there, but it might be sensible to rename it to something like a Teerex Api and place it there.. but I honestly don't have a full overview of the current worker folder structure. @clangenb might be the right person here to ask?

Reply:
My comment is already outdated. In the meantime, @clangenb and I have already discussed these next steps. The sending of the extrinsic is now part of the light client (so no new crate). This can be seen in the next PR ☺️